vyper/builtins/functions.py

```diff
@@ -1267,6 +1267,8 @@ def infer_arg_types(self, node, expected_return_typ=None):
 
     @process_inputs
     def build_IR(self, expr, args, kwargs, context):
+        context.check_is_not_constant(f"use {self._id}", expr)
```

Comment:
ideally i'd like this to be in the frontend or even use `BuiltinFunctionT._modifiability`, but this works for now